apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/stu3/providers/PatientResourceProvider.java

```diff
@@ -459,25 +472,62 @@ private Beneficiary selectBeneWithLatestReferenceYear(List<Beneficiary> duplicat
     return entityManager.find(Beneficiary.class, maxReferenceYearMatchingBeneficiaryId);
   }
 
-  /** Enumerates the supported "should we include unique beneficiary identifiers" options. */
-  public static enum IncludeIdentifiersMode {
-    INCLUDE_HICNS_AND_MBIS,
+  /**
+   * The header key used to determine which header should be used. See {@link
+   * #returnIncludeIdentifiersValues(RequestDetails)} for details.
+   */
+  public static final String HEADER_NAME_INCLUDE_IDENTIFIERS = "IncludeIdentifiers";
 
-    OMIT_HICNS_AND_MBIS;
+  /**
+   * The List of valid values for the {@link #HEADER_NAME_INCLUDE_IDENTIFIERS} header. See {@link
+   * #returnIncludeIdentifiersValues(RequestDetails)} for details.
+   */
+  public static final List<String> VALID_HEADER_VALUES_INCLUDE_IDENTIFIERS =
+      Arrays.asList("true", "false", "hicn", "mbi");
 
-    /**
-     * The header key used to determine which {@link IncludeIdentifiersMode} mode should be used.
-     * See {@link #determineIncludeIdentifiersMode(RequestDetails)} for details.
-     */
-    public static final String HEADER_NAME_INCLUDE_IDENTIFIERS = "IncludeIdentifiers";
-
-    static IncludeIdentifiersMode determineIncludeIdentifiersMode(RequestDetails requestDetails) {
-      String includeIdentifiersValue = requestDetails.getHeader(HEADER_NAME_INCLUDE_IDENTIFIERS);
-      if (Boolean.parseBoolean(includeIdentifiersValue) == true) {
-        return INCLUDE_HICNS_AND_MBIS;
-      } else {
-        return OMIT_HICNS_AND_MBIS;
-      }
-    }
+  /**
+   * Return a valid List of values for the IncludeIdenfifiers header
+   *
+   * @param requestDetails a {@link RequestDetails} containing the details of the request URL, used
+   *     to parse out include identifiers values
+   * @return List of validated header values against the {@link
+   *     VALID_HEADER_VALUES_INCLUDE_IDENTIFIERS} list.
+   */
+  public static List<String> returnIncludeIdentifiersValues(RequestDetails requestDetails) {
+    String headerValues = requestDetails.getHeader(HEADER_NAME_INCLUDE_IDENTIFIERS);
+
+    if (headerValues == null || headerValues == "") return Arrays.asList("");
+    else
+      // Return values split on a comma with any whitespace, valid, distict, and sort
+      return Arrays.asList(headerValues.toLowerCase().split("\\s*,\\s*")).stream()
+          .peek(
+              c -> {
+                if (!VALID_HEADER_VALUES_INCLUDE_IDENTIFIERS.contains(c))
+                  throw new InvalidRequestException(
+                      "Unsupported " + HEADER_NAME_INCLUDE_IDENTIFIERS + " header value: " + c);
+              })
+          .distinct()
+          .sorted()
+          .collect(Collectors.toList());
+  }
```

Comment:
Why do you need the `Arrays.asList`? Doesn't `split().stream()...collect(Collect.toList())` return a `List<String>`?

Reply:
The Arrays.asList() encloses just `headerValues.toLowerCase().split("\\s*,\\s*")` that produces a String[] array which doesn't have a steam() method.

Reply:
Ok.